Add a class to pass SeqNr and snapshot payload in

rtar opened the pull request:
The main question here is if we want to have `payload` as an `Option` or require it always to be filled.

This idea of `Option` was taken from a similar `Event` case class, but I am not sure it applies the same way to a snapshot.

Z1kkurat commented:
I struggle to imagine a case where we'd need a snapshot without a state inside. It looks like a tombstone, but we don't need those for persistent storage such as Cassandra, do we? We either have a snapshot with a state, or we delete the row altogether

rtar commented:
@Z1kkurat  I copied from `Event` structure.

I now suspect it was added there as part of the https://github.com/evolution-gaming/kafka-journal/commit/c6d0543b908b1475e8a6bdf4eebc45d5d8733dd5 commit to ensure backwards compatibility with the previously stored binary snapshot.

I will rework it to remove `Option` now.

CC: @dfakhritdinov

rtar commented:
I have removed `Option` now (and also removed it from a large PR, which I am deconstructing into these smaller PRs, but you do not have to look there: https://github.com/evolution-gaming/kafka-journal/pull/532).
